src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -194,26 +187,23 @@ type TableContext<'TRecord> internal
     let queryAsync (keyCondition : ConditionalExpr.ConditionalExpression)
                     (filterCondition : ConditionalExpr.ConditionalExpression option)
                     (projectionExpr : ProjectionExpr.ProjectionExpr option)
-                    (limit: int option) (consistentRead : bool option) (scanIndexForward : bool option)
-                    collector = async {
+                    (limit: int option)
+                    (consistentRead : bool option)
+                    (scanIndexForward : bool option) = async {
 
-        let! (downloaded, _) = queryPaginatedAsync keyCondition filterCondition projectionExpr (LimitType.AllOrCount limit) None consistentRead scanIndexForward collector
+        let! (downloaded, _) = queryPaginatedAsync keyCondition filterCondition projectionExpr (LimitType.AllOrCount limit) None consistentRead scanIndexForward
 
         return downloaded
     }
 
     let scanPaginatedAsync (filterCondition : ConditionalExpr.ConditionalExpression option)
                             (projectionExpr : ProjectionExpr.ProjectionExpr option)
-                            (limit : LimitType) (exclusiveStartKey : TableKey option) (consistentRead : bool option)
-                            collector = async {
+                            (limit : LimitType) (exclusiveStartKey : TableKey option) (consistentRead : bool option) = async {
 
         let downloaded = new ResizeArray<_>()
-        let returnConsumedCapacity, maybeReport = metricsOptions collector
         let consumedCapacity = new ResizeArray<ConsumedCapacity>()
         let emitMetrics () =
-            match maybeReport with
-            | None -> ()
-            | Some r -> r Scan (Seq.toList consumedCapacity) downloaded.Count
+            maybeReport |> Option.iter (fun r -> r Scan (Seq.toList consumedCapacity) downloaded.Count)
```

Comment:
the List rendition does not gain anyone much here IMO
I'd tend to use just a `seq` in the record definition
Having said all of that, for the typical count of 1, a `list` is a perfectly efficient way to render it so it doesn't really matter a lot either...

Reply:
I'm tempted to remove the list altogether and just emit a new metrics call every AWS call. Or alternatively just do a better job translating & interpreting `ConsumedCapacity` as the AWS response tends to be confusing and unexpected if you don’t know exactly what it should return for each operation

Reply:
I note also it always asks for indexed, which might be redundant.
Regarding your point about the response being confusing, I am seeing zeros from https://github.com/jet/equinox/blob/ddb/src/Equinox.DynamoStore/DynamoStore.fs#L297 so assumed the simulator was returning zeros - this tells me I'm assuming wrong! (I'd have put a breakpoint in a test and looked at the values in this repo if I knew how to use Expecto though!)

Reply:
Regarding triggering multiple calls, I think one per roundtrip triggered is appropriate and useful - e.g. for a query loop, it allows one to tag each roundtrip response with a timestamp which can make things clearer if you are doing large queries under load. I'd be for that.

Reply:
I can’t find the original reference now, but I was also seeing values not populated where I expected them to be, and found a quote from a random internet person suggesting some of them are only populated on certain operations, and some were going to be populated in future but are not done yet. I’ve only ended up using `CapacityUnits`, perhaps it’s best just to surface those for now and add more later if there’s demand?

Reply:
Hm, I was explicitly separating the read charges from the writes in Equinox as that will likely be pretty critical context for things like optimising `TransactWriteItems` calls, but even a boring conditional Put or Get already benefits from the breakdown - only surfacing a single value would not be ideal unless its unavoidable (and even in that case, I'd be looking for this lib to smooth out such wrinkles rather than concealing context)

Reply:
Ideally yes, the current design is more a result of me avoiding doing that work. I suspect this might end up needing experimentation to understand exactly what’s returned in what scenario.

Reply:
I might leave the existing `ConsumedCapacity list` model for the moment while I ponder how I might go about translating the responses into something more useful - purely to avoid changing again.